### ISSUE TITLE
Improve error message on unresolved dependencies

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#1422]
## steps

Something missing in transitive dependency due to people deleting artifacts from artifactory or depending on SNAPSHOT or whatnot. 
## problem

@paulp wrote in #381:

>  Dammit Jim, I'm a programmer, not a hexadecimal scanner.

```
[info] Loading project definition from /scratch/trunk1/project/project
[info] Updating {file:/scratch/trunk1/project/project/}default-2ac879...
... snip 38 lines of resolving ...
[info] Done updating.
Cloning into '/Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23'...
[info] Loading project definition from /Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23/project
[info] Updating {file:/Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23/project/}default-9f14df...
[info] Resolving org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1 ...
[warn]  module not found: org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1

... blah blah lots more resolving
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[error] {file:/Users/paulp/.sbt/0.12.0-M1/staging/eff7d7476f58ab859d23/project/}default-9f14df/*:update: sbt.ResolveException: unresolved dependency: org.scala-tools.sbt#scripted-plugin_2.9.1;0.12.0-M1: not found
```
## expectation

sbt should provide as much info on where the missing dependency is coming from in plain English, and possibly advise the user on what to do/read/check next.

**Comment by jsuereth** [comment]
+1.  I believe on failure the ivy partial graph is available for inspection.
